app/models/payment_gateway_setting.rb

```diff
@@ -37,4 +39,25 @@ def active_gateway_type
       errors.add(:gateway_type, :invalid)
     end
   end
+
+  def increment_failure_count
+    lock!
+    incremented_failure_count = failure_count + 1
+    self.gateway_settings = symbolized_settings.merge({failure_count:  incremented_failure_count})
```

Comment:
I was thinking of using ActiveRecord::Store as a refactoring

`store :gateway_settings, accessors: some_stuffs`